README.md

```diff
@@ -45,25 +47,122 @@ Documentation can be generated with [ExDoc](https://github.com/elixir-lang/ex_do
+```
+
+Check if your NervesKey has been provisioned:
+
+```elixir
+iex> NervesKey.provisioned?(i2c)
+true
+```
+
+If you get `false`, go to the provisioning sections. If you received an error,
+check that the NervesKey has a good connection to your hardware. If you have a
+custom board, you may need to pass parameters to
+`ATECC508A.Transport.I2C.init/1` to set the correct I2C bus.
+
+NervesKeys are provisioned with serial numbers. In production, these can be of
+your choosing.
+
+```elixir
+iex> NervesKey.manufacturer_sn(i2c)
+"ABC12345"
+```
+
+Of course, the more interesting part of the NervesKeys are its storage of device
+private keys and their certificates. For the common case, it stores two X.509
+certificates: one for the device and one for the certificate that signed the device
+certificate. The signer certificate is usually uploaded to the servers that the
+device will connect to so that it can authenticate the device. Here's how
+to get both of the certificates:
+
+```elixir
+iex> NervesKey.device_cert(i2c)
+{:OTPCertificate, ...}
+
+# Put this in a convenient form:
+iex> X509.Certificate.to_pem(v()) |> IO.puts
+-----BEGIN CERTIFICATE-----
+stuff
+stuff
+stuff
+-----END CERTIFICATE-----
+
+iex> NervesKey.signer_cert(i2c)
+{:OTPCertificate, ...}
+```
+
+The next step is to tell Erlang's SSL library that you want to use the NervesKey
+when connecting to the server. For that, you'll need [nerves_key_pkcs11](https://github.com/nerves-hub/nerves_key_pkcs11). This code is somewhat tedious
+but hopefully the following code fragment will help:
+
+```elixir
+   {:ok, engine} = NervesKey.PKCS11.load_engine()
+   {:ok, i2c} = ATECC508A.Transport.I2C.init([])
+
+   signer_cert = X509.Certificate.to_der(NervesKey.signer_cert(i2c))
+   cert = X509.Certificate.to_der(NervesKey.device_cert(i2c))
+   key = NervesKey.PKCS11.private_key(engine, {:i2c, 1})
+   cacerts = [signer_cert] ++ Keyword.get(opts, :trusted_certs, [])
+
+   Tortoise.Supervisor.start_child(
+     server: {
+       Tortoise.Transport.SSL,
+       verify: :verify_peer,
+       host: Keyword.get(opts, :host),
+       cert: cert,
+       key: key,
+       cacerts: cacerts,
+       versions: [:"tlsv1.2"],
+     })
+```
+
 ## Preparing for provisioning
 
-The ATECC508A in the NervesKey needs to be provisioned before it can be used.
-Before you can do that, you'll need a signing certificate and some information
-about your device.
+The ATECC508A/608A in the NervesKey needs to be provisioned before it can be
+used.  Before you can do that, you'll need the following:
+
+1. A signing certificate and its private certificate (in other contexts, this is
+   called a certificate authority)
+2. A serial number for your device
+3. A name for the device
+
+The signing certificate and serial number of very important. After the
+provisioning process, they are locked down and cannot be changed without
+replacing the ATECC508A. The device name is purely informational unless you
```

Comment:
Should we always list ATECC508A and ATECC608A when referring to the chips? or maybe ATECC508/608A ?